File: Server/route/image.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetImageRoute(r *gin.Engine) {
	image := r.Group("/image")
	image.Use(middleware.AdminAuthMiddleware())
	{
		image.POST("/GetImageList", api.GetImageList)
		image.POST("/GetImage", api.GetImage)
		image.POST("/CreateImage", api.CreateImage)
		image.POST("/UpdateImage", api.UpdateImage)
		image.POST("/DeleteImage", api.DeleteImage)
	}
}

File: Web/src/api/image.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetImageList(data) {
    return apiClient.post("/image/GetImageList", data).then(handleResponse).catch(handleError)
}

function GetImage(data) {
    return apiClient.post("/image/GetImage", data).then(handleResponse).catch(handleError)
}

function CreateImage(data) {
    return apiClient.post("/image/CreateImage", data).then(handleResponse).catch(handleError)
}

function UpdateImage(data) {
    return apiClient.post("/image/UpdateImage", data).then(handleResponse).catch(handleError)
}

function DeleteImage(data) {
    return apiClient.post("/image/DeleteImage", data).then(handleResponse).catch(handleError)
}

const imageApi = {
  GetImageList,
  GetImage,
  CreateImage,
  UpdateImage,
  DeleteImage
}

export default imageApi
